core/src/wallet/algorand/AlgorandBlockchainExplorer.cpp

```diff
@@ -0,0 +1,200 @@
+                    const auto& json = std::get<1>(response)->GetObject();
+                    auto account = model::Account();
+                    JsonParser::parseAccount(json, account);
+                    return account;
+            });
+    }
+
+    Future<model::AssetParams> BlockchainExplorer::getAssetById(uint64_t id) const
+    {
+        return _http->GET(fmt::format(constants::purestakeAssetEndpoint, id))
+            .json(false)
+            .map<model::AssetParams>(
+                    getContext(),
+                    [id](const HttpRequest::JsonResult& response) {
+                        auto assetParams = model::AssetParams();
+                        const auto& json = std::get<1>(response)->GetObject();
+                        if (!json.HasMember(constants::xAsset.c_str())) {
+                            throw make_exception(api::ErrorCode::CURRENCY_NOT_FOUND,
+                                                 fmt::format("Asset {} does not exist", id));
+                        }
+                        const auto& jsonParams =
+                            json[constants::xAsset.c_str()].GetObject()[constants::xParams.c_str()].GetObject();
+                        JsonParser::parseAssetParams(jsonParams, assetParams);
+                        return assetParams;
+                    });
+    }
+
+    Future<model::Transaction>
+    BlockchainExplorer::getTransactionById(const std::string & txId) const {
+        return _http->GET(fmt::format(constants::purestakeTransactionEndpoint, txId))
+            .json(false)
+            .map<model::Transaction>(getContext(), [&txId](const HttpRequest::JsonResult& response) {
+                    const auto& jsonArray = std::get<1>(response)->GetObject()[constants::xTransactions.c_str()].GetArray();
+                    if (jsonArray.Size() != 1) {
+                        throw make_exception(api::ErrorCode::TRANSACTION_NOT_FOUND,
+                                             fmt::format("Couldn't find transaction {}", txId));
+                    }
+                    const auto& json = *std::begin(jsonArray);
+                    auto tx = model::Transaction();
+                    JsonParser::parseTransaction(json, tx);
+                    return tx;
+            });
+    }
+
+    Future<model::TransactionsBulk>
+    BlockchainExplorer::getTransactionsForAddress(const std::string & address,
+                                                  const Option<uint64_t> & firstRound,
+                                                  const Option<uint64_t> & lastRound) const
+    {
+        auto url = fmt::format(constants::purestakeAccountTransactionsEndpoint, address);
+        url = fmt::format("{}?limit={}", url, constants::EXPLORER_QUERY_LIMIT);
+        if (firstRound) {
+            url = fmt::format("{}&min-round={}", url, *firstRound);
+        }
+        if (lastRound) {
+            url = fmt::format("{}&max-round={}", url, *lastRound);
+        }
```

Comment:
Please turn these string pattern into constant variable (since all the other are actually)